stdlib/bytes.gr

```diff
@@ -49,561 +61,656 @@ let checkIndexIsInBounds = (i, byteSize, max) => {
+  let size = getSize(src)
+  let leftOrig = left
+  let rightOrig = right
+  let left = coerceNumberToWasmI32(left)
+  let right = coerceNumberToWasmI32(right)
+  let newSize = size + left + right
+  if (newSize < 0n) {
+    throw Exception.InvalidArgument("The resulting length is less than 0")
+  }
+  let dst = allocateBytes(newSize)
+  let mut srcOffset = 0n
+  let mut dstOffset = 0n
+  if (left < 0n) {
+    srcOffset = left * -1n
+    dstOffset = 0n
+  }
+  if (left > 0n) {
+    srcOffset = 0n
+    dstOffset = left
+  }
+  let len = if (right < 0n) {
+    size + right - srcOffset
+  } else {
+    size - srcOffset
+  }
+  if (len > 0n) {
+    // formatter-ignore
+    Memory.copy(
+      dst + _VALUE_OFFSET + dstOffset,
+      src + _VALUE_OFFSET + srcOffset,
+      len,
+    )
+  }
+  let ret = WasmI32.toGrain(dst): (Bytes)
+  Memory.decRef(WasmI32.fromGrain(leftOrig))
+  Memory.decRef(WasmI32.fromGrain(rightOrig))
+  Memory.decRef(WasmI32.fromGrain(bytes))
+  Memory.decRef(WasmI32.fromGrain(resize))
   ret
 }
 
 /**
- * Gets a signed 32-bit integer starting at the given byte index.
- * @param i: Number - The byte index
- * @param b: Bytes - The byte sequence
- * @returns Int32
+ * Copies a range of bytes from a source byte sequence to a given location
+ * in a destination byte sequence.
+ *
+ * @param srcIndex: The starting index to copy bytes from
+ * @param dstIndex: The starting index to copy bytes into
+ * @param length: The amount of bytes to copy from the source buffer
+ * @param src: The source byte sequence
+ * @param dst: The destination byte sequence
+ *
+ * @since v0.3.2
  */
 @disableGC
-export let rec getInt32 = (i: Number, b: Bytes) => {
+export let rec move =
+  (
+    srcIndex: Number,
+    dstIndex: Number,
+    length: Number,
+    src: Bytes,
+    dst: Bytes,
+  ) => {
+  let (>) = WasmI32.gtS
   let (+) = WasmI32.add
-  let ptr = WasmI32.fromGrain(b)
-  let size = getSize(ptr)
-  let offset = coerceNumberToWasmI32(i)
-  checkIndexIsInBounds(offset, _INT32_BYTE_SIZE, size)
-  let n = WasmI32.load(ptr + offset, _VALUE_OFFSET)
-  let ret = Conv.toInt32(n)
-  Memory.decRef(WasmI32.fromGrain(i))
-  Memory.decRef(WasmI32.fromGrain(b))
-  Memory.decRef(WasmI32.fromGrain(getInt32))
+  let srcIndexOrig = srcIndex
+  let dstIndexOrig = dstIndex
+  let lenthOrig = length
+  let src = WasmI32.fromGrain(src)
+  let srcSize = getSize(src)
+  let srcIndex = coerceNumberToWasmI32(srcIndex)
+  let dst = WasmI32.fromGrain(dst)
+  let dstSize = getSize(dst)
+  let dstIndex = coerceNumberToWasmI32(dstIndex)
+  let length = coerceNumberToWasmI32(length)
+  if (srcIndex + length > srcSize) {
+    throw Exception.InvalidArgument("Invalid source bytes range")
+  }
+  if (dstIndex + length > dstSize) {
+    throw Exception.InvalidArgument("Invalid destination bytes range")
+  }
+  let end = srcIndex + length
+  // formatter-ignore
+  let ret = Memory.copy(
+    dst + _VALUE_OFFSET + dstIndex,
+    src + _VALUE_OFFSET + srcIndex,
+    length,
+  )
```

Comment:
cc @marcusroberts - this broke crazily without the comment:
```
let ret = Memory.copy(
  dst + _VALUE_OFFSET +
  dstIndex,
  src + _VALUE_OFFSET +
  srcIndex,
  length,
)
```

Reply:
I think that's fixed in #959 

marcus@Marcuss-Air grain % grain format simple.gr
let ret = Memory.copy(
  dst + _VALUE_OFFSET +
  dstIndex,
  src + _VALUE_OFFSET +
  srcIndex,
  length,
)

assuming that's what you needed?

Reply:
Ignore my last comment, yes that's still there in the latest formatter,  so raising issue #965

Reply:
Fixed in #966